src/backend/distributed/commands/utility_hook.c

```diff
@@ -292,6 +294,21 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		PG_RE_THROW();
 	}
 	PG_END_TRY();
+
+	if (AddAllLocalTablesToMetadata && ShouldConvertTable && !IsBinaryUpgrade &&
+		IsA(parsetree, CreateStmt) && context == PROCESS_UTILITY_TOPLEVEL &&
+		IsCoordinator() && CoordinatorAddedAsWorkerNode())
+	{
+		CreateStmt *createTableStmt = (CreateStmt *) parsetree;
+		if (createTableStmt->relation->relpersistence != 't' && createTableStmt->partbound == NULL)
+		{
+			ShouldConvertTable = false;
+			CommandCounterIncrement();
+			Oid relationId = RangeVarGetRelid(createTableStmt->relation, NoLock, false); 
+			CreateCitusLocalTable(relationId, true, false);
+			ShouldConvertTable = true;
```

Comment:
now that we have PROCESS_UTILITY_TOPLEVEL check, do we still need ShouldConvertTable?

This isn't a very reliable flag at the moment, e.g. errors may leave the flag as false, but I'm wondering whether we need it.